### PR TITLE
Convert Name to a class shared by Member and Event

### LINKED ISSUE #107: Make Name a shared class
- Move Name and any other associated files to a `shared` folder in `model`
- Rename fullName attribute to name
- Delete EventName and associated files
- Refactor all imports of Name

This is to prevent duplicate work when parsing member names and event names as parsing of both are exactly the same.

### PR DESCRIPTION
**Pull request by dylkaw**
Closes #107 with all its stated changes.

### CODE REVIEW

File: src/main/java/seedu/address/model/event/EventNameContainsKeywordsPredicate.java [hunk 1]
```diff
@@ -6,8 +6,9 @@
 import seedu.address.commons.util.StringUtil;
 import seedu.address.commons.util.ToStringBuilder;
 
+
 /**
- * Tests that a {@code Event}'s {@code Name} matches any of the keywords given.
+ * Tests that a {@code Member}'s {@code Name} matches any of the keywords given.
```
Comment (wr1159) [inline src/main/java/seedu/address/model/event/EventNameContainsKeywordsPredicate.java]: Do you think we can remove the extra line? Also is this meant to be Event?

File: src/main/java/seedu/address/model/event/EventNameContainsKeywordsPredicate.java [hunk 2]
```diff
@@ -33,9 +34,9 @@ public boolean equals(Object other) {
             return false;
         }
 
-        EventNameContainsKeywordsPredicate otherEventNameContainsKeywordsPredicate =
+        EventNameContainsKeywordsPredicate otherMemberNameContainsKeywordsPredicate =
                 (EventNameContainsKeywordsPredicate) other;
-        return keywords.equals(otherEventNameContainsKeywordsPredicate.keywords);
+        return keywords.equals(otherMemberNameContainsKeywordsPredicate.keywords);
```
Comment (wr1159) [inline src/main/java/seedu/address/model/event/EventNameContainsKeywordsPredicate.java]: Similarly, I think this should be kept as otherEventNameContainsKeywordsPredicate

**Review by wr1159 — COMMENTED** [review]
LGTM aside from 2 small naming concerns

**Review by SelwynAng — COMMENTED** [review]
LGTM!

**Pull request merged**